bbb.js

```diff
@@ -1,77 +1,48 @@
-/*eslint-disable no-var*/ /*eslint-disable vars-on-top*/ /*eslint-disable no-shadow*/
-var video = document.querySelector("video");
-var assetURL = "frag_copy.mp4";
-var mimeCodec = "video/mp4; codecs=\"avc1.64001f, mp4a.40.2\"";
-var mediaSource = new MediaSource();
-var appendButton = document.getElementById("append");
-var seekButton = document.getElementById("seek");
+const assetURL = "frag_copy.mp4";
+const video = document.querySelector("video");
+const mimeCodec = "video/mp4; codecs=\"avc1.64001f, mp4a.40.2\"";
+const mediaSource = new MediaSource();
+const seekButton = document.getElementById("seek");
 
-mediaSource.addEventListener("sourceopen", onSourceOpen.bind(this));
+mediaSource.addEventListener("sourceopen", onSourceOpen);
 video.src = window.URL.createObjectURL(mediaSource);
 
-async function onSourceOpen(e){
+async function onSourceOpen(){
   console.log("onSourceOpen");
-  var mediaSource = e.target;
 
-  var sourceBuffer = mediaSource.addSourceBuffer(mimeCodec);
-  // sourceBuffer.mode = "sequence";
-  // sourceBuffer.addEventListener("updateend", updateEnd);
+  const sourceBuffer = mediaSource.addSourceBuffer(mimeCodec);
+  const initSegment = await GetSegment("0-2265771"); // Append the initialization segment & first media segment
+  sourceBuffer.appendBuffer(initSegment);
```

Comment:
usually init segments and media are done in two separate appends. I only did init and first media segment for simplicity with what you originally had.